fellow-submissions/Cielo-Raymundo.md

```diff
@@ -0,0 +1,138 @@
+    if it is being done inside it's scope. Var hoisting variables allows the declaration to be read the declration before it is even declared and
+    will set its value to undefined, while let and const variables don't get hoisted meaning that they aren't read before they are declared, 
+    they are read when they are declared.  
+
+**2. What will the following code log? Why?**
+    ```javascript
+    let a = 'outer';
+
+    function testScope() {
+      let a = 'inner';
+      console.log(a);
+    }
+
+    console.log(a);
+    testScope();
+    console.log(a);
+    ```
+    This will console log :
+    outer
+    inner
+    outer
+    
+    Because the variable a creates a block scope surrounding the whole program, so when a is called the value that it has is "outer", but inside
+    the function testScope it is declared and initialized again but this doesn't affect the first variable since this one only exists inisde the
+    function. So when it is called inisde the function the value is "inner".
+**3. What will the following code log? Why?**
+    ```javascript
+    function hello() {
+      let a = 'hello';
+    }
+
+    hello();
+    console.log(a);
+    ```
+    This will console log:
+    undefined
+    a is undefined
+    
+    Because the function isn't returning anything so it's just undefined. Since the variable a only exists inside the function hello, it can't 
+    be accessed outside the function/scope. 
+**4. What does the following code log? Why?**
+    ```javascript
+    console.log(a);
+
+    var a = 1;
+    ```
+   This will console log:
+   Undefined
+   
+   Because even though the variable is being hoisted, only the declaration is being read so it isn't initialized until it is fully compiled. 
+   
+**5. What does the following code log? Why?**
+    ```javascript
+    console.log(a);
+
+    let a = 1;
+    ```
+    This will console log:
+    a is undefined
+    
+    Because let variables don't get hoisted, they don't exist before they are declared. So the variable a is something that doesn't exist
+    
+**6. In the code snippet below, the function declaration comes _after_ the variable declaration in the code below. Why does it still work?**
+    ```javascript
+    let name = "Devonte";
+
+    function sayItLoud() {
+      console.log(`${name}!!!`)
+    }
+
+    sayItLoud();
+    ```
+    Because name is a variable who's blockscope includes the function sayItLoud, so when that function uses the variable name, it will
+    take the value it was assigned to.
+    
+**7. Why does the following code throw an error? What is the key difference between it and the example above?**
+    ```javascript
+    sayItLoud();
+
+    function sayItLoud() {
+      console.log(`${greeting}!!!}`)
+    }
+
+    let greeting = "Hello"
+    ```
+    This doesn't work because the function sayItLoud doesn't exist inside the variables greeting's scope so greeting inside the function doesn't
```

Comment:
Incorrect here. The scopes are the same as the question above. The _order_ of declaration/function invocation has changed though. Run this in the console and tell me what you get.